socket/server.go

```diff
@@ -77,7 +88,10 @@ func (s *DefaultServer) Listen() error {
 
 // handleClient handles a client that has been accepted from the server.
 func (s *DefaultServer) handleClient(c *Client) {
-	defer c.Close(s.serverRegistry)
+	defer s.handleClientDisconnect(c)
+	s.clientsMu.Lock()
+	s.clients[c.Name()] = c
```

Comment:
At this stage, the client hasn't sent any packet other than initializing a TCP connection, so the name is empty. It'd make sense to move this to the auth request handler.